contracts/FastEcMul.sol

```diff
@@ -443,7 +443,13 @@ library FastEcMul {
   /// @param _x the integer
   /// @return y the square root of _x
   function _sqrt(uint256 _x) private pure returns (uint256) {
-    uint256 z = (_x + 1) / 2;
+    uint256 z;
+    // Separate in two cases to avoid overflow
+    if (_x + 1 == 0) {
```

Comment:
This way to check for overflow relies on overflow to happen. If the purpose of this PR is to fix the case where `_x` is the max value, then this is fine. But if the purpose is to silent warnings about overflow, try something like this instead:

 ```suggestion
     if (_x == u256_max_value) {
```

Reply:
Maybe this?
```suggestion
    if (_x + 1 < x) {
```

Reply:
Updated!

Reply:
As suggested by auditors, use @tmpolaczyk's solution, hardcoding the result for X=0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF and doing a test for it